implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -622,7 +624,7 @@ private HealthCheckResponse.Status fireEventIfStatusChanged(Event<HealthStatusCh
         return newStatus;
     }
 
-    private void fireGlobalHealthStatusChangeIfNeeded() {
+    private void fireGlobalHealthStatusChangeIfNeeded(SmallRyeHealth smallRyeHealth) {
 
         HealthCheckResponse.Status newStatus = UP;
         if (List.of(livenessStatus, readinessStatus, startupStatus, wellnessStatus).contains(DOWN)) {
```

Comment:
This is probably not strictly related to this PR, but should these four statuses be rather passed as arguments from the lambda on line 591, rather than read from class fields? I could imagine that due to a race condition, their value may have changed by the time you get here from lines 577-580.

Reply:
(probably very unlikely that it will cause a problem, but it caught my attention when reading this code)